staging/kos/pkg/controllers/connectionagent/connectionagent.go

```diff
@@ -0,0 +1,1091 @@
+		// case (len(state.localAtts) == 0), the vni is no longer relevant to the connection agent, hence
+		// we return the vnState after removing it from the map storing all the vnStates so that the caller
+		// can perform the necessary clean up. If that's not the case, we add the attachment name to the remote
+		// attachments in the vnState.
+		if vnState != nil {
+			vnState.remoteAtts[attName] = struct{}{}
+			delete(vnState.localAtts, attName)
+			if len(vnState.localAtts) == 0 {
+				delete(ca.vniToVnState, vni)
+				vnStateRet = vnState
+				return
+			}
+		}
+	}
+
+	return
+}
+
+func (ca *ConnectionAgent) updateVNStateAfterAttDeparture(attName string, vni uint32) bool {
+	vnState := ca.removeAttFromVNState(attName, vni)
+	if vnState == nil {
+		return false
+	}
+	// If we're here attName was the last local attachment in the virtual network with id vni. Hence we
+	// stop the remote attachments informer and enqueue references to remote attachments in that virtual
+	// network, so that their interfaces can be deleted.
+	ca.stopRemoteAttInformerAndEnqueueRemoteAttRefs(vnState, vni)
+	return true
+}
+
+func (ca *ConnectionAgent) createOrUpdateIfc(attState *attachmentState,
+	attGuestIP, attHostIP gonet.IP,
+	attVNI uint32,
+	attNSN k8stypes.NamespacedName) (existingIfc netfabric.NetworkInterface, attHasIfc bool, err error) {
+
+	if attState != nil {
+		existingIfc, attHasIfc = attState.ifc, attState.ifcIsValid
+	}
+
+	newIfcNeedsToBeCreated := !attHasIfc || ifcNeedsUpdate(existingIfc, attGuestIP, attHostIP, attVNI)
+
+	err = ca.deleteIfc(existingIfc, attHasIfc && newIfcNeedsToBeCreated)
+	if err != nil {
+		err = fmt.Errorf("Update of network interface of attachment %s failed, old interface %#+v could not be deleted: %s",
+			attNSN,
+			existingIfc,
+			err.Error())
+		return
+	}
+
+	if newIfcNeedsToBeCreated {
+		attHasIfc = false
+		guestMAC := generateMACAddr(attVNI, attGuestIP)
+		newIfc := netfabric.NetworkInterface{
+			Name:     generateIfcName(guestMAC),
+			VNI:      attVNI,
+			GuestMAC: guestMAC,
+			GuestIP:  attGuestIP,
+			HostIP:   attHostIP,
+		}
+		if attHostIP.Equal(ca.hostIP) {
+			err = ca.netFabric.CreateLocalIfc(newIfc)
+		} else {
+			err = ca.netFabric.CreateRemoteIfc(newIfc)
+		}
+		if err != nil {
+			err = fmt.Errorf("Creation of network interface of attachment %s failed, interface %#+v could not be created: %s",
+				attNSN,
+				newIfc,
+				err.Error())
+			return
+		}
+		existingIfc = newIfc
+		attHasIfc = true
+	}
+
+	return
+}
+
+func (ca *ConnectionAgent) deleteIfc(ifc netfabric.NetworkInterface, ifcNeedsDeletion bool) error {
+	var err error
+	if ifcNeedsDeletion {
+		if ifc.HostIP.Equal(ca.hostIP) {
+			// If we're here the interface is local
+			err = ca.netFabric.DeleteLocalIfc(ifc)
+		} else {
+			// If we're here the interface is remote
+			err = ca.netFabric.DeleteRemoteIfc(ifc)
+		}
+	}
+	return err
+}
+
+func (ca *ConnectionAgent) setHostIPAndIfcNameInLocalAttStatus(att *netv1a1.NetworkAttachment,
```

Comment:
This name is overly specific.  There is only one function here that updates an attachment's status.  This could be called something simpler like `setAttStatus`.

Reply:
Fixed in 8672bd4.